spynnaker/pyNN/models/neuron/population_machine_synapses.py

```diff
@@ -0,0 +1,479 @@
+# Copyright (c) 2017-2020The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import ctypes
+from collections import namedtuple
+
+from spinn_utilities.overrides import overrides
+from spinn_utilities.abstract_base import abstractproperty
+
+from spinn_front_end_common.utilities.helpful_functions import (
+    locate_memory_region_for_placement)
+from spinn_front_end_common.abstract_models import (
+    AbstractSupportsBitFieldGeneration,
+    AbstractSupportsBitFieldRoutingCompression)
+from spinn_front_end_common.interface.provenance import (
+    ProvidesProvenanceDataFromMachineImpl)
+from spinn_front_end_common.utilities.utility_objs import ProvenanceDataItem
+
+from spynnaker.pyNN.models.neuron.synapse_dynamics import (
+    AbstractSynapseDynamicsStructural)
+from spynnaker.pyNN.utilities.utility_calls import get_n_bits
+from spynnaker.pyNN.utilities import bit_field_utilities
+from spynnaker.pyNN.models.abstract_models import AbstractSynapseExpandable
+
+from .synaptic_matrices import SynapticMatrices
+
+
+get_placement_details = \
```

Comment:
VOM! this is a static method. why not just make it public and import properly?!

Reply:
The methods should definitely be made public here